### ISSUE TITLE
Panel doesn't synchronize with the text

### ISSUE THREAD

**Issue by feinstein** [mohzy83/NppMarkdownPanel#23]
As I type new text, the Panel stays the same, I have to close it and open again so it will show me the new text.

**Comment by vinsworldcom** [comment]
Yes, me too.  Notepad++ 7.8.6 64-bit.  Looking at the code:

NppMarkdownPanel\NppMarkdownPanel\MarkdownPanelController.cs
49:	        public void OnNotification(ScNotification notification)

traps some events, in fact, you trap the right event:

NppMarkdownPanel\NppMarkdownPanel\MarkdownPanelController.cs
```
66:  else if (notification.Header.Code == (uint)SciMsg.SCN_MODIFIED)
```

But the checking for SC_MOD_INSERTTEXT or SC_MOD_DELETETEXT seems to always turn up false as it never gets to 

```
74:  RenderMarkdown();
```

I commented out:

NppMarkdownPanel\NppMarkdownPanel\MarkdownPanelController.cs
```
68:	                    // bool isInsert = (notification.ModificationType & (uint)SciMsg.SC_MOD_INSERTTEXT) != 0;
69:	                    // bool isDelete = (notification.ModificationType & (uint)SciMsg.SC_MOD_DELETETEXT) != 0;
70:	                    // // Any modifications made ?
71:	                    // if (isInsert || isDelete)
72:	                    // {
73:	                        lastTickCount = Environment.TickCount;
74:	                        RenderMarkdown();
75:	                    // }
```

and it started automatically refreshing as I typed!

I also added a bit of scrolling buffer so that the Markdown preview doesn't sync it's top line with the N++ buffer with "Synchronize viewer with caret position" enabled.  Basically, I just adjust down 10 lines, but it's configurable in the .INI file (maybe you want to expose that in your settings dialogue?).

C# is not my language; I was very lucky this compiled with no errors on VS2017.  The following is a patch including my only changes:

----<CLIP HERE>----
```
diff --git a/NppMarkdownPanel/MarkdownPanelController.cs b/NppMarkdownPanel/MarkdownPanelController.cs
index fe87cf5..30ef5d9 100644
--- a/NppMarkdownPanel/MarkdownPanelController.cs
+++ b/NppMarkdownPanel/MarkdownPanelController.cs
@@ -35,6 +35,7 @@ namespace NppMarkdownPanel
         private int lastCaretPosition;
         private string iniFilePath;
         private bool syncViewWithCaretPosition;
+        private int ScrollBuffer;

         public MarkdownPanelController()
         {
@@ -55,7 +56,7 @@ namespace NppMarkdownPanel
                     if (syncViewWithCaretPosition && lastCaretPosition != scintillaGateway.GetCurrentPos().Value)
                     {
                         lastCaretPosition = scintillaGateway.GetCurrentPos().Value;
-                        ScrollToElementAtLineNo(scintillaGateway.GetCurrentLineNumber());
+                        ScrollToElementAtLineNo(scintillaGateway.GetCurrentLineNumber() - ScrollBuffer);
                     }
                 }
                 else
@@ -65,14 +66,14 @@ namespace NppMarkdownPanel
                 }
                 else if (notification.Header.Code == (uint)SciMsg.SCN_MODIFIED)
                 {
-                    bool isInsert = (notification.ModificationType & (uint)SciMsg.SC_MOD_INSERTTEXT) != 0;
-                    bool isDelete = (notification.ModificationType & (uint)SciMsg.SC_MOD_DELETETEXT) != 0;
-                    // Any modifications made ?
-                    if (isInsert || isDelete)
-                    {
+                    // bool isInsert = (notification.ModificationType & (uint)SciMsg.SC_MOD_INSERTTEXT) != 0;
+                    // bool isDelete = (notification.ModificationType & (uint)SciMsg.SC_MOD_DELETETEXT) != 0;
+                    // // Any modifications made ?
+                    // if (isInsert || isDelete)
+                    // {
                         lastTickCount = Environment.TickCount;
                         RenderMarkdown();
-                    }
+                    // }
                 }
             }
         }
@@ -117,7 +118,8 @@ namespace NppMarkdownPanel
             SetIniFilePath();
             syncViewWithCaretPosition = (Win32.GetPrivateProfileInt("Options", "SyncViewWithCaretPosition", 0, iniFilePath) != 0);
             markdownPreviewForm.CssFileName = Win32.ReadIniValue("Options", "CssFileName", iniFilePath, "style.css");
-            markdownPreviewForm.ZoomLevel = Win32.GetPrivateProfileInt("Options", "ZoomLevel", 130, iniFilePath);
+            markdownPreviewForm.ZoomLevel = Win32.GetPrivateProfileInt("Options", "ZoomLevel", 100, iniFilePath);
+            ScrollBuffer = Win32.GetPrivateProfileInt("Options", "ScrollBuffer", 10, iniFilePath);
             PluginBase.SetCommand(0, "About", ShowAboutDialog, new ShortcutKey(false, false, false, Keys.None));
             PluginBase.SetCommand(1, "Toggle Markdown Panel", TogglePanelVisible);
             PluginBase.SetCommand(2, "Synchronize viewer with caret position", SyncViewWithCaret, syncViewWithCaretPosition);
@@ -169,6 +171,7 @@ namespace NppMarkdownPanel
         public void PluginCleanUp()
         {
             Win32.WritePrivateProfileString("Options", "SyncViewWithCaretPosition", syncViewWithCaretPosition ? "1" : "0", iniFilePath);
+            Win32.WriteIniValue("Options", "ScrollBuffer", ScrollBuffer.ToString(), iniFilePath);
             SaveSettings();
         }
```
----<CLIP HERE>----

Cheers.

**Comment by mohzy83** [comment]
this is fixed in release 0.6.1
